pennylane/ops/op_math/controlled.py

```diff
@@ -204,6 +224,62 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+@functools.lru_cache  # only create the first time requested
+def _get_ctrl_qfunc_prim():
+    """See capture/explanations.md : Higher Order primitives for more information on this code."""
+    # if capture is enabled, jax should be installed
+    import jax  # pylint: disable=import-outside-toplevel
+
+    ctrl_prim = jax.core.Primitive("ctrl_transform")
+    ctrl_prim.multiple_results = True
+
+    @ctrl_prim.def_impl
+    def _(*args, n_control, jaxpr, control_values, work_wires, n_consts):
+        consts = args[:n_consts]
+        args = args[n_consts:]
+        control_wires = args[-n_control:]
+        args = args[:-n_control]
+
+        with qml.queuing.AnnotatedQueue() as q:
+            jax.core.eval_jaxpr(jaxpr, consts, *args)
+        ops, _ = qml.queuing.process_queue(q)
+
+        for op in ops:
+            ctrl(op, control_wires, control_values, work_wires)
+        return []
```

Comment:
I think here we agreed on `None` right?
```suggestion
```

Reply:
The primitives themselves can't return None, as there's no "abstract none".  But the user-facing `qml.ctrl` will return `None` in this case.